synapse/handlers/presence.py

```diff
@@ -383,16 +381,15 @@ def bump_presence_active_time(self, user):
 
         bump_active_time_counter.inc()
 
-        prev_state = yield self.current_state_for_user(user_id)
+        prev_state = await self.current_state_for_user(user_id)
 
         new_fields = {"last_active_ts": self.clock.time_msec()}
         if prev_state.state == PresenceState.UNAVAILABLE:
             new_fields["state"] = PresenceState.ONLINE
 
-        yield self._update_states([prev_state.copy_and_replace(**new_fields)])
+        await self._update_states([prev_state.copy_and_replace(**new_fields)])
 
-    @defer.inlineCallbacks
-    def user_syncing(self, user_id, affect_presence=True):
+    async def user_syncing(self, user_id, affect_presence=True):
```

Comment:
I'd be tempted to stick a type annotation on this one, since I find it hard to get straight and mypy could save us from some footgunnery. I think the correct return type is `typing_extensions.ContextManager[None]`.